spec/lib/submission_builder/ty2024/states/nj/documents/nj1040_spec.rb

```diff
@@ -497,6 +497,55 @@
       end
     end
 
+    describe "Other Retirement Income Exclusion - line 28a" do
+      before do
+        allow(Flipper).to receive(:enabled?).with(:show_retirement_ui).and_return(true)
+      end
+
+      context "when filer is eligible" do
+        it 'sets PensionExclusion to the line 28a value' do
+          expected = 5_000
+          allow_any_instance_of(Efile::Nj::Nj1040Calculator).to receive(:calculate_line_28a).and_return expected
+          expect(xml.at("PensionExclusion").text).to eq(expected.to_s)
+        end
+      end
+    end
+
+    describe "Other Retirement Income Exclusion - line 28b" do
+      before do
+        allow(Flipper).to receive(:enabled?).with(:show_retirement_ui).and_return(true)
+      end
+      
+      context "when filer is eligible" do
+        it 'sets OtherRetireIncomeExclus to the line 28b value' do
+          expected = 1_000
+          allow_any_instance_of(Efile::Nj::Nj1040Calculator).to receive(:calculate_line_28b).and_return expected
+          expect(xml.at("OtherRetireIncomeExclus").text).to eq(expected.to_s)
+        end
+      end
+
+      context "when filer is ineligible" do
```

Comment:
[dust] Could we change the labels here to "when 28b has a value / does not have a value" or something similar? Nothing here is really dealing with the concept of eligibility, that's handled and tested elsewhere.

Reply:
Same for the other tests in this file above and below